core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2424,21 +2446,54 @@ public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFile
     }
     if (options.getOwner() != null) {
       inode.setUserName(options.getOwner());
+      if (inode.isPersisted()) {
+        MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
+        String ufsUri = resolution.getUri().toString();
+        UnderFileSystem ufs = resolution.getUfs();
+        try {
+          ufs.setOwner(ufsUri, inode.getUserName(), inode.getGroupName());
+        } catch (IOException e) {
+          throw new AccessControlException("Could not setOwner for UFS file " + ufsUri, e);
+        }
+      }
     }
     if (options.getGroup() != null) {
       inode.setGroupName(options.getGroup());
+      if (inode.isPersisted()) {
+        MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
+        String ufsUri = resolution.getUri().toString();
+        UnderFileSystem ufs = resolution.getUfs();
+        try {
+          ufs.setOwner(ufsUri, inode.getUserName(), inode.getGroupName());
+        } catch (IOException e) {
+          throw new AccessControlException("Could not setGroup for UFS file " + ufsUri, e);
+        }
+      }
     }
     if (options.getPermission() != Constants.INVALID_MODE) {
```

Comment:
same as above, once #3566 is merged, there will no longer be confusion between `permission` and `mode`.